src/Valleysoft.DockerCredsProvider.Test/CredsProviderTests.cs

```diff
@@ -9,11 +9,19 @@ namespace Valleysoft.DockerCredsProvider.Test;
 
 public class CredsProviderTests
 {
+    IEnvironment DefaultEnvironmentMock;
+
+    public CredsProviderTests() {
+        var envMock = new Mock<IEnvironment>();
+        envMock.Setup(e => e.GetUserProfilePath()).Returns(new Environment().GetUserProfilePath());
+        DefaultEnvironmentMock = envMock.Object;
```

Comment:
I went with a unified mock for the environment because every test needed one branch or the other - either the environment variable needed to return, or the default user profile path needed to be returned.  This made it so that most tests were using the 'default' behavior, and the one test that covered the environment-variable logic could make that change locally, instead of changing the global shared state of the process' environment variables.